subsys/bluetooth/controller/ll_sw/ull_scan.c

```diff
@@ -151,9 +161,56 @@ uint8_t ll_scan_enable(uint8_t enable)
 		}
 	}
 
-#if defined(CONFIG_BT_CTLR_PRIVACY)
-	struct lll_scan *lll;
+#if defined(CONFIG_BT_CTLR_ADV_EXT)
+	lll = &scan->lll;
+	if (duration) {
+		lll->duration_reload = ((uint32_t)duration * 10000U / 625U) /
+					scan->lll.interval;
+		if (period) {
+			if (IS_ENABLED(CONFIG_BT_CTLR_PARAM_CHECK) &&
+			    (duration >= ((uint32_t)period << 7))) {
+				return BT_HCI_ERR_INVALID_PARAM;
+			}
+
+			scan->duration_lazy = ((uint32_t)period * 1280000U /
+					       625U) / scan->lll.interval;
+			scan->duration_lazy -= lll->duration_reload;
+			scan->node_rx_scan_term = NULL;
+		} else {
+			struct node_rx_pdu *node_rx_scan_term;
+			void *link_scan_term;
+
+			scan->duration_lazy = 0U;
+
+			/* The alloc here used for ext scan termination event */
+			link_scan_term = ll_rx_link_alloc();
+			if (!link_scan_term) {
+				/* TODO: figure out right return value */
+				return BT_HCI_ERR_MEM_CAPACITY_EXCEEDED;
```

Comment:
I would argue that `BT_HCI_ERR_MEM_CAPACITY_EXCEEDED` sounds like a proper return value, and the TODO (both this and the one below) should be removed.

Reply:
What about insufficient resources?

Reply:
That would also be a good alternative. However, it seems that `BT_HCI_ERR_MEM_CAPACITY_EXCEEDED` is typically returned when `ll_rx_xxx_alloc()` fails. See also `ull_adv.c:740`.